### ISSUE TITLE
Longitudinal EEG Model Ladder: identical splits, calibration budgets, and transfer authority

### ISSUE THREAD

**Issue by sidhulyalkar** [sidhulyalkar/neurOS-v1#26]
## Objective

Build the first **method-comparison layer** on top of the longitudinal evidence contract in PR #25.

The benchmark must answer:

> Given identical prior-session history, identical labeled target-session calibration examples, and one immutable final evaluation set, which decoder/representation strategy most improves next-session performance and reduces calibration burden?

No method may choose its own split, consume future sessions under a prospective claim, or inspect final evaluation examples for preprocessing, source weighting, adaptation, model selection, or early stopping.

## Authority

The split/calibration manifests produced by the longitudinal evidence layer are the authority:

- `history_policy=prior` by default;
- exact source-session identities;
- target session;
- calibration indices at each budget;
- immutable evaluation indices;
- partition/calibration fingerprints;
- dataset/Git/package identity.

Every model result row must carry those fingerprints.

### A. Transparent floor

- CSP + LDA

Already implemented by #25.

### B. neurOS supervised task decoders

- EEGNet
- EEG-Conformer

Use the existing `neuros-models` PyTorch implementations and their real `train / predict_proba / encode / analysis_manifest` contracts.

Initial comparison should train each architecture from the same source-history + declared calibration indices at each budget. Hyperparameters must be fixed before final evaluation.

Record:

- model configuration;
- seed;
- training history;
- model/manifest fingerprint;
- fit wall-clock;
- inference latency;
- balanced accuracy / ROC-AUC;
- optional calibration metrics.

### C. Frozen representation + matched readout

For each inspectable encoder:

1. train/freeze the encoder using allowed source-history data only;
2. extract embeddings from allowed source-history and calibration examples;
3. fit one matched low-capacity readout;
4. evaluate on frozen target-session evaluation embeddings.

This separates representation transfer from end-to-end fine-tuning.

### D. SourceWeigher on frozen representations

Use the *same frozen encoder* for all source sessions and the target calibration set so all domain summaries live in one representation space.

Each prior session is a candidate source domain:

```text
S1 embeddings --+
S2 embeddings --+--> RepresentationSourceWeigher --> source weights
S3 embeddings --+
                                      ^
                                      |
                         target calibration embeddings only
```

Then fit a matched readout on source embeddings with per-example weights derived from source-session weights, plus the declared target calibration samples.

Do **not** compare embeddings produced by separately trained source-specific encoders; their coordinates are not guaranteed to be aligned.

Do **not** estimate target moments from final evaluation examples.

## Zero-calibration semantics

Target-dependent SourceWeigher is **not defined at labeled calibration budget 0** unless a separate unlabeled target-observation protocol is introduced.

Do not silently use evaluation examples as unlabeled target observations and call that zero-shot.

If an unlabeled adaptation experiment is added later, report two independent budgets:

- labeled calibration examples;
- unlabeled target observation examples / seconds.

## Foundation-model lane

Foundation representations enter the ladder only through verified adapters that expose a genuine representation/encoding capability.

Current guardrails:

- ZUNA reconstruction is not an embedding baseline and reconstructed samples are not measurements;
- catalog-only models do not count as executable evidence;
- NeuroFM-X remains experimental until a pinned trained artifact and stable encode contract exist;
- add faithful upstream EEG representation adapters rather than copying architectures or returning placeholders.

## Metrics

Primary:

- balanced accuracy / task-appropriate official metric;
- performance-vs-calibration frontier;
- calibration examples required to reach predeclared target performance;
- area under calibration frontier.

Secondary:

- ROC-AUC where appropriate;
- expected calibration error / Brier score;
- fit/adaptation wall-clock;
- inference latency;
- failed-run rate;
- effective representation rank / anisotropy;
- session leakage probe;
- CKA / representation stability across sessions;
- SourceWeigher entropy / ESS / max source weight / residual.

## Statistical comparison

Use paired subject-session cases and preserve failures.

Before a public comparison:

- predeclare primary metric;
- predeclare subject/session set;
- use multiple model seeds for neural decoders;
- report paired uncertainty across subject-session cases;
- do not tune against final evaluation examples;
- retain skipped/failed cases in the artifact.

## Artifact schema

Extend the evidence bundle with method identity rather than creating separate result formats:

```text
study_manifest.json
splits.json or embedded split manifests
methods/<method-id>/method_manifest.json
results.csv
summary.json
report.md
artifact_hashes.json
```

Each result row should minimally include:

```text
method_id
model_seed
subject
held_out_session
history_policy
calibration_per_class
partition_fingerprint
calibration_split_fingerprint
balanced_accuracy
roc_auc
fit_s
inference_ms_per_trial
status / failure_reason
```

## Evidence Console

The main visualization becomes a **paired calibration frontier**:

```text
performance
^
|                     EEG-Conformer + SourceWeigher
|               *-----*
|          *----       EEG-Conformer
|      *---            EEGNet
|  *---                CSP+LDA
+--------------------------------> labeled calibration / class
  0      1      2       5       10
```

Selecting a point must reveal:

- exact subject/session case;
- source sessions;
- calibration examples consumed;
- SourceWeigher weights if applicable;
- model/manifest identity;
- partition/evaluation fingerprint;
- runtime and uncertainty.

## Implementation sequence

1. Merge/qualify PR #25.
2. Add a method-runner protocol around frozen longitudinal manifests.
3. Implement EEGNet + EEG-Conformer under identical indices.
4. Add frozen-encoder + matched-readout mode.
5. Add SourceWeigher target-calibration-only weighting.
6. Add paired statistics / failure preservation.
7. Execute a small Kumar2024 pilot evidence bundle.
8. Inspect the artifact before scaling subjects/seeds.
9. Scale the predeclared study.
10. Render the same bundle in the Evidence Console.

## Promotion rule

A method belongs on the public neurOS calibration frontier only if its result can be traced back to the exact same split/calibration authority as every competing method.

**Comment by sidhulyalkar** [comment]
## Kumar2024-specific constraints added by preregistration #27

The model ladder should consume Issue #27 as part of its evidence authority.

In particular:

- Kumar2024 has two original training/adaptation cohorts that must remain visible: MOABB subjects 1–9 = GR, subjects 10–18 = PAR.
- The promoted comparison should not treat the resulting 90 subject-session cases as independent observations; use subject-clustered/hierarchical inference and report GR/PAR-specific estimates.
- Primary method-comparison endpoint is the paired area under the balanced-accuracy calibration frontier at the predeclared budgets `0,1,2,5,10`.
- Pilot split seed is `2026`; promoted split seeds are predeclared in #27 rather than selected after model results are visible.
- A small provenance pilot should include at least one GR and one PAR participant (recommended `1,10`) before all-18-subject scale-up.
- The current upstream Kumar2024 loader explicitly maps MOABB session keys `0..5` to the chronological offline + five online sessions, so this dataset can use a verified dataset-specific prospective chronology rather than an inferred lexical ordering.

No model-ladder implementation should weaken these constraints.

**Comment by sidhulyalkar** [comment]
## Proposed v1 model-ladder freeze after model-contract audit

To avoid tuning the benchmark after seeing Kumar2024 results, the first neural comparison should use the current neurOS architecture defaults unless a change is justified *before* the real-data run.

### EEGNet v1

Use current `EEGNetModel` defaults at the dataset sampling rate:

```text
temporal_filters=8
depth_multiplier=2
separable_filters=16
temporal_kernel=63
separable_kernel=15
dropout=0.25
learning_rate=1e-3
weight_decay=1e-4  # TorchDecoderModel default
n_epochs=20
batch_size=32
```

For 22-channel, 2-class Kumar2024 this is roughly 1.5k trainable parameters. Record the exact dynamic parameter count in the method artifact rather than relying on this estimate.

### EEG-Conformer v1

Use current `EEGConformerModel` defaults:

```text
embedding_dim=40
temporal_kernel=25
pool_length=25
pool_stride=10
n_heads=4
n_layers=4
feedforward_multiplier=4
dropout=0.30
learning_rate=3e-4
weight_decay=1e-4
n_epochs=20
batch_size=32
```

For 22-channel, 2-class Kumar2024 this is roughly 117k trainable parameters, ~80x EEGNet. Therefore any public comparison must report capacity/fit time/latency and must not narrate the result as a clean test of 'attention vs convolution'. These are intentionally different-capacity reference architectures.

### Training semantics

Current `TorchDecoderModel` uses fixed epochs and seeded shuffling with no hidden validation/early-stopping set. Preserve that for v1: it avoids creating another data-selection surface that could leak target evaluation information.

Predeclare model seeds; recommended first comparison:

```text
101, 503, 1601
```

Split seeds remain governed separately by #27.

Current code seeds PyTorch/CUDA RNGs but does not globally force deterministic CUDA algorithms. Before promoted GPU evidence, either:

1. add/qualify an explicit deterministic-training mode and record it in the artifact, or
2. treat model seed as optimization randomness and report multi-seed uncertainty rather than claiming bitwise reproducibility.

Do not silently equate `random_state` with full GPU determinism.

### Input/preprocessing

For the first ladder, keep the same MOABB 8-30 Hz task trials and native source sampling used by the CSP floor unless we deliberately version a new common-resampling protocol and rerun **all** methods including CSP. Do not compare a resampled deep-model protocol against a native-rate CSP artifact as if only the model changed.

Both current deep models contain normalization layers; avoid adding a method-specific target normalization path unless it is defined train-only and applied consistently.

### Frozen-representation lane

For each target session:

1. train the encoder once on prior-session source history only;
2. freeze it;
3. compute all source-session embeddings in that one coordinate system;
4. compute target calibration embeddings for each allowed budget;
5. fit a matched low-capacity readout on allowed data;
6. evaluate the same frozen target evaluation examples.

Target calibration must not fine-tune the encoder in this lane. End-to-end target fine-tuning is a separate method.

### SourceWeigher v1

Use the same frozen encoder described above.

For calibration-limited targets, make the initial representation summary **mean-only** (`statistics=("mean",)`) rather than mean + log standard deviation. At 1/class the target distribution contains only two samples, making spread estimates extremely unstable even though the current implementation is numerically defined.

A later `mean+log_std` SourceWeigher variant can be a separately named method with a predeclared minimum target-observation budget.

SourceWeigher diagnostics to persist per case/budget:

```text
source-session weights
ESS
entropy
max source weight
moment residual
source distances
condition number
convergence / iterations
```

At labeled calibration budget 0, target-dependent SourceWeigher stays unavailable unless we explicitly add an independently budgeted unlabeled-target protocol.

**Comment by sidhulyalkar** [comment]
This internal model-ladder roadmap is now superseded by the production NSQ external-method authority and the frozen Kumar2024 study tracked in #82/#90.

The important scientific ideas here survive, but they should no longer be implemented through a privileged neurOS-only ladder that can see different internal surfaces than an external method. Current policy is:

- methods enter public comparison through the same NSQ factory / case / calibration / final-assessment authority;
- SourceWeigher and ORION must use that same boundary when eventually compared;
- the frozen promoted Kumar2024 external floor is MNE CSP+LDA, pyRiemann RG+LR, and three EEGNet model seeds;
- ORION/frozen-representation/SourceWeigher experiments come only after that external floor is executed and audited.

Closing this as superseded prevents contributors from implementing a second comparison authority. Useful representation/SourceWeigher metrics in this issue remain candidate future analyses under the current NSQ/ORION evidence program.
